File: projects/chapter12_inheritance/main/scope_test/Main.java
public class Main {
    public static void main(String[] args) {
        int scope_num = 0;
        for(int i = 0; i < 5; i++){
            if(i == 4) {
                scope_num = 1;
            }
            System.out.println(scope_num);
        }
        //Testing if scope_num is accessible here
        System.out.println("Final scope_num: " + scope_num);
    }
}

File: projects/chapter12_inheritance/main/scope_test/main.cpp
#include <iostream>
using namespace std;

int main() {
    int scope_num = 0;
    for(int i = 0; i < 5; i++){
        if(i == 4) {
            int scope_num = 2;
            cout << scope_num << endl;
        }
        cout << scope_num << endl;
    }
    //Testing if scope_num is accessible here
    cout << "Final scope_num: " << scope_num << endl;
    return 0;
}
